buyer.go

```diff
@@ -156,12 +157,22 @@ func newTicketPurchaser(cfg *config,
 		maintainMinPrice: maintainMinPrice,
 		useMedian:        cfg.FeeSource == useMedianStr,
 		priceMode:        priceMode,
+		heightCheck:      make(map[int64]struct{}),
 	}, nil
 }
 
 // purchase is the main handler for purchasing tickets for the user.
 // TODO Not make this an inlined pile of crap.
 func (t *ticketPurchaser) purchase(height int64) error {
+
+	// Check to make sure that the current height has not already been seen
+	// for a reorg or a fork
+	if _, exists := t.heightCheck[height]; exists {
+		return fmt.Errorf("We've already seen this height, "+
```

Comment:
This should be tracef, then just return nil. Handling a reorganization is not an error.